logical/framework/path_map_test.go

```diff
@@ -254,6 +254,36 @@ func TestPathMap_Salted(t *testing.T) {
 	if v != nil {
 		t.Fatalf("bad: %#v", v)
 	}
+
+	// Put in a non-salted version and make sure that after reading it's been
+	// upgraded
+	err = storage.Put(&logical.StorageEntry{
+		Key:   "struct/map/foo/b",
+		Value: []byte(`{"foo": "bar"}`),
+	})
+	if err != nil {
+		t.Fatal("err: %v", err)
+	}
+	// A read should transparently upgrade
+	resp, err = b.HandleRequest(&logical.Request{
+		Operation: logical.ReadOperation,
+		Path:      "map/foo/b",
+		Storage:   storage,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	list, _ := storage.List("struct/map/foo/")
```

Comment:
Should you check the length of list here to make sure only one result is returned (testing the delete worked)